sdk/go/common/resource/plugin/analyzer_plugin.go

```diff
@@ -222,7 +222,7 @@ func (a *analyzer) AnalyzeStack(resources []AnalyzerStackResource) ([]AnalyzeDia
 				continue
 			}
 
-			pdeps := []string{}
+			pdeps := make([]string, 0, 1)
```

Comment:
This seems odd to me, there are cases where pdeps will be empty. This would make this line always allocate a slice?

Reply:
In line 221 https://github.com/pulumi/pulumi/blob/dab81be2624df4f4de031a4884deadfd4ef1db1e/sdk/go/common/resource/plugin/analyzer_plugin.go#L221

The program checked lenth of `pd`. The length of `pd` must be greater than 1.